### PR TITLE
Issue1314 - update join_backfill guide

### LINKED ISSUE #1314: Join Backfill Kapacitor Issues
https://docs.influxdata.com/kapacitor/v1.3/guides/join_backfill/

Some things I had trouble with while going through the example:

- There are scripts to generate example data but they are only linked to at the bottom. They should be at the top so that users can follow along with their own
- The script to generate data are horribly out of date and do not work on influxdb 1.3.x. the influx_stress command is out of date and even when you update to use the `-config` option, it segfaults
- We should mention in the `groupBy` part of the description how join handles tags. I had issues with tag values not matching exactly and my join was not working.

### PR DESCRIPTION
**Pull request by karel-rehor**
Updating join_backfill.md to resolve #1314.  Adding a Python3 script to generate the data discussed in the document. 

Documentation staged on AWS:

http://ec2-54-153-106-230.us-west-1.compute.amazonaws.com:1414/kapacitor/v1.5/guides/join_backfill/

### CODE REVIEW

File: content/kapacitor/v1.4/guides/join_backfill.md [hunk 1]
```diff
@@ -9,23 +9,54 @@ menu:
     parent: Guides
 ---
 
-Often times we have set of series where each series is counting a particular event.
-Using Kapacitor we can join those series and calculate a combined value.
+Collecting a set of data series where each data series is counting a particular event is a common scenario.
```
Comment (stevebang) [inline content/kapacitor/v1.4/guides/join_backfill.md]: Replace "data series" with "time series".

File: content/kapacitor/v1.4/guides/join_backfill.md [hunk 2]
```diff
@@ -9,23 +9,54 @@ menu:
...
+Using Kapacitor, multiple data series in a set can be joined and used to calculate a combined value, which can then be stored as a new data series.
```
Comment (stevebang) [inline content/kapacitor/v1.4/guides/join_backfill.md]: Replace "data series" with "time series".

File: content/kapacitor/v1.5/guides/join_backfill.md [hunk 3]
```diff
@@ -5,28 +5,58 @@ aliases:
...
     name: Calculating rates across series
-    identifier: join_backfill
-    weight: 1
+    weight: 10
     parent: guides
 ---
 
-Often times we have set of series where each series is counting a particular event.
-Using Kapacitor we can join those series and calculate a combined value.
+Collecting a set of data series where each data series is counting a particular event is a common scenario.
```
Comment (stevebang) [inline content/kapacitor/v1.5/guides/join_backfill.md]: Replace "data series" with "time series".

File: content/kapacitor/v1.5/guides/join_backfill.md [hunk 4]
```diff
@@ -5,28 +5,58 @@ aliases:
...
+Using Kapacitor, multiple data series in a set can be joined and used to calculate a combined value, which can then be stored as a new data series.
```
Comment (stevebang) [inline content/kapacitor/v1.5/guides/join_backfill.md]: Replace "data series" with "time series".

File: content/kapacitor/v1.4/guides/join_backfill.md [hunk 5]
```diff
@@ -9,23 +9,54 @@ menu:
...
 
-Let's say we have two measurements:
+This guide will show how to use a prepared data generator in python and then how
+to combine two generated data series into a new calculated measurement and then
+store that measurement back into InfluxDB, all through Kapacitor.
```
Comment (sanderson) [inline content/kapacitor/v1.4/guides/join_backfill.md]: Switch to present tense. Some suggested structure changes:

> This guide shows how to use a prepared data generator in Python
> to combine two generated data series into a new calculated measurement, then
> store that measurement in InfluxDB using Kapacitor.

File: content/kapacitor/v1.4/guides/join_backfill.md [hunk 6]
```diff
@@ -9,23 +9,54 @@ menu:
...
-Then to store the data back into InfluxDB as a new measurement.
 
 ### Joining with batch data
 
-We need to query the two measurements, `errors` and `views`.
+Having simple counts may not be sufficient for a site administrator. More
+important would be to know the percent of page views that are resulting in error.
+The process is to select both existing measurements, to join them and then to
+calculate an error percentage.  The error percentage can then be stored back into
+InfluxDB as a new measurement.
```
Comment (sanderson) [inline content/kapacitor/v1.4/guides/join_backfill.md]: Structure adjustments:

> The process is to select both existing measurements, join them and calculate an error percentage.
> The error percentage can then be stored in InfluxDB as a new measurement.

File: content/kapacitor/v1.4/guides/join_backfill.md [hunk 7]
```diff
@@ -9,23 +9,54 @@ menu:
...
+
+The two measurements, `errors` and `views`, will need to be queried.
```
Comment (sanderson) [inline content/kapacitor/v1.4/guides/join_backfill.md]: Change to present tense: "will need" -> "need"

File: content/kapacitor/v1.4/guides/join_backfill.md [hunk 8]
```diff
@@ -58,14 +94,15 @@ errors
         .as('errors', 'views')
 ```
 
-The data is joined by time, meaning that as pairs of batches arrive from each source they will be combined into a single batch.
-As a result the fields from each source need to be renamed to properly namespace the fields.
-This is done via the `.as('errors', 'views')` line.
-In this example each measurement has only one field named `sum`,
-the joined fields will be called `errors.sum` and `views.sum` respectively.
+The data is joined by time, meaning that as pairs of batches arrive from each source
+they will be combined into a single batch. As a result the fields from each source
```
Comment (sanderson) [inline content/kapacitor/v1.4/guides/join_backfill.md]: Change to present tense: "will be" -> "are"

File: content/kapacitor/v1.4/guides/join_backfill.md [hunk 9]
```diff
@@ -58,14 +94,15 @@ errors
...
+need to be renamed to properly namespace the fields. This is done via the
+`.as('errors', 'views')` line.  In this example each measurement has only one field
+named `sum`.  The joined fields will be called `errors.sum` and `views.sum` respectively.
```
Comment (sanderson) [inline content/kapacitor/v1.4/guides/join_backfill.md]: Change to present tense.

File: content/kapacitor/v1.5/guides/join_backfill.md [hunk 10]
```diff
@@ -5,28 +5,58 @@ aliases:
...
 
-Let's say we have two measurements:
+This guide will show how to use a prepared data generator in python and then how
```
Comment (sanderson) [inline content/kapacitor/v1.5/guides/join_backfill.md]: Same as above. Change the present tense, capitalize "python", and update the structure of the sentence.

File: content/kapacitor/v1.4/guides/join_backfill.md (hunk 1 above)
Comment (karel-rehor) [reply content/kapacitor/v1.4/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.4/guides/join_backfill.md (hunk 2 above)
Comment (karel-rehor) [reply content/kapacitor/v1.4/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.5/guides/join_backfill.md (hunk 3 above)
Comment (karel-rehor) [reply content/kapacitor/v1.5/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.5/guides/join_backfill.md (hunk 4 above)
Comment (karel-rehor) [reply content/kapacitor/v1.5/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.4/guides/join_backfill.md (hunk 5 above)
Comment (karel-rehor) [reply content/kapacitor/v1.4/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.4/guides/join_backfill.md (hunk 6 above)
Comment (karel-rehor) [reply content/kapacitor/v1.4/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.4/guides/join_backfill.md (hunk 7 above)
Comment (karel-rehor) [reply content/kapacitor/v1.4/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.4/guides/join_backfill.md (hunk 8 above)
Comment (karel-rehor) [reply content/kapacitor/v1.4/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.4/guides/join_backfill.md (hunk 9 above)
Comment (karel-rehor) [reply content/kapacitor/v1.4/guides/join_backfill.md]: fixed

File: content/kapacitor/v1.5/guides/join_backfill.md (hunk 10 above)
Comment (karel-rehor) [reply content/kapacitor/v1.5/guides/join_backfill.md]: Fixed along with other changes required in v1.4, which have been copied to v1.5

**Comment by sanderson** [comment]
@karel-rehor Looks like there are some merge conflicts in this PR. Once those are resolved, I'll go ahead and merge.

**Comment by karel-rehor** [comment]
I've reviewed and fixed the conflicts

**Comment by karel-rehor** [comment]
I've updated the section on Stream and added a new section on `kapacitor record query`.

Please review.

**Pull request merged**